server/src/lobby.rs

```diff
@@ -180,6 +173,32 @@ impl Actor for Lobby {
     }
 }
 
+impl Lobby {
+    fn send_filtered_positions(&self, vhcs: Vec<Vehicle>) {
+        self.clients.keys().for_each(|client_id| {
+            if let Some(client) = self.clients.get(client_id) {
+                if let Some(client_pos) = &client.position {
+                    let filtered_vhcs = vhcs
+                        .clone()
+                        .into_iter()
+                        .filter(|vhc| filter_vehicle_position(client_pos, vhc))
+                        .collect::<Vec<Vehicle>>();
+                    self.send_message(
+                        &serde_json::to_string(&ServerOutput::VehiclePositions(
+                            VehiclePositionsOutput {
+                                timestamp: Lobby::get_current_timestamp(),
+                                vehicles: filtered_vhcs,
+                            },
+                        ))
+                        .unwrap(),
+                        client_id,
+                    );
```

Comment:
Not entirely sure, but if the `filtered_vhcs` vector is empty after the filter a message is sent regardless if the client can actually "see" anything on their map. We should add a check to see if the `filtered_vhcs` is empty or not and only send data if it contains something.